avi/utils.go

```
@@ -630,3 +633,22 @@ func IsPostNotAllowed(objtype string) bool {
 	}
 	return specialobj
 }
+
+func update_cluster_state(d *schema.ResourceData, meta interface{}, s map[string]*schema.Schema) error {
+	client := meta.(*clients.AviClient)
+	var err error
+	var robj interface{}
+	err = client.AviSession.Get("api/cluster/runtime", &robj)
+	if err == nil {
+		if local_data, err := SchemaToAviData(d, s); err == nil {
+			mod_api_res, err := SetDefaultsInAPIRes(robj, local_data, s)
+			if err != nil {
+				log.Printf("[ERROR] Update Cluster State in modifying api response object %v\n", err)
+			}
+			if _, err := ApiDataToSchema(mod_api_res, d, s); err != nil {
+				log.Printf("[ERROR] Converting ApiDataToSchema object %v\n", err)
```

this error is not returned back.

This code needs to be refactored to properly propagate the errors.